plugins/org.jboss.tools.openshift.express.core/src/org/jboss/tools/openshift/express/internal/core/server/ExpressServerUtils.java

```diff
@@ -132,33 +133,64 @@ public static IProject getDeployProject(IServerAttributes attributes) {
 	 * Look-up the OpenShift application associated with the given server. This
 	 * operation can be time-consuming since it may need to perform a request on
 	 * OpenShift if the user's applications list had not been loaded before.
-	 * Callers should use this methd without blocking the UI.
+	 * Callers should use this method without blocking the UI.
+	 * 
+	 * Never returns null, throws GetApplicationException or OpenShiftException 
+	 * if by any reason the OpenShift application cannot be found.
+	 * That will help ui clients to display problem in a user-friendly way.
 	 * 
 	 * @param server
 	 *            the server
-	 * @return the openshift application or null if it could not be located.
-	 * @throws OpenShiftException
+	 * @return the openshift application or throw exception if it could not be located.
+	 * @throws OpenShiftException, GetApplicationException
 	 */
-	public static IApplication getApplication(IServerAttributes server) {
+	public static IApplication getApplication(IServerAttributes server) throws GetApplicationException {
 		final String appName = getApplicationName(server);
 		if (StringUtils.isEmpty(appName)) {
-			return null;
+			throw new GetApplicationException("No application.");
 		}
 		final ConnectionURL connectionUrl = getConnectionUrl(server);
 		if (connectionUrl == null) {
-			return null;
+			throw new GetApplicationException(NLS.bind("Failed to get connection URL from {0}", server.getName()));
+		}
+		ExpressConnection connection = null;
+		String expectedConnectionProblem = NLS.bind("Failed to find connection {0}", connectionUrl.toString());
+		try {
+			connection = ConnectionsRegistrySingleton.getInstance().getByUrl(connectionUrl, ExpressConnection.class);
+			if (connection != null) {
+				if(!connection.connect()) {
+					throw new GetApplicationException(NLS.bind("Connection {0} is not authenticated.", connectionUrl.toString()));
+				}
+			} else {
+				throw new GetApplicationException(expectedConnectionProblem);
+			}
+		} catch (OpenShiftException e) {
+			ExpressCoreActivator.pluginLog().logError(expectedConnectionProblem, e);
+			throw new GetApplicationException(expectedConnectionProblem);
```

Comment:
Then, we have to show it. I suggest that if action catches GetApplicationException with a cause inside, it will not show a message dialog, but rethrow the cause to be shown in standard error dialog.